pwngdb.py

```diff
@@ -605,15 +605,14 @@ def getfmtarg(addr):
     if capsize == 0 :
         getarch()
     if arch == "i386" :
-        start = get_reg("esp")
+        start = get_reg("esp") + 4
         idx = (addr- start)/4
-        print("The index of format argument : %d" % idx)
     elif arch == "x86-64" :
-        start = get_reg("rsp")
+        start = get_reg("rsp") + 8
         idx = (addr - start)/8 + 6
-        print("The index of format argument : %d" % idx)
     else :
         print("Not support the arch")
+    print("The index of format argument : %d (\"\%%%d$p\")" % (idx,idx))
```

Comment:
There's a 'else' case so this shouldn't be moved to here

Reply:
thanks!you are right,I will correct it